src/ExtUvLoop.php

```diff
@@ -313,4 +313,31 @@ private function createStreamListener()
 
         return $callback;
     }
+
+    /**
+     * @param float $interval
+     * @return int
+     */
+    private function convertFloatSecondsToMilliseconds($interval)
+    {
+        if ($interval < 0) {
+            throw new \InvalidArgumentException('Interval cannot be lower than 0.');
+        }
+
+        // Subtract 2 because 1 doesn't seems to work and may give wrong results
+        $maxValue = ((int) (\PHP_INT_MAX / 1000)) - 2;
```

Comment:
This calculation looks really *odd*, i.e. it doesn't explain *why* or *what* "wrong results" could be. As an alternative, what do you think about something like this:

```php
$result = $interval * 1000;
if ($result > PHP_INT_MAX) {
    // nope
}
return (int)$result;
```

Reply:
Your code will work only if `$result` will be about ~1025 more than `PHP_INT_MAX`,
check this example: https://3v4l.org/OH3T7

Reply:
You're right. PHP will automatically convert an `int` on overflow to a `float`, but `PHP_INT_MAX + 1 > PHP_INT_MAX === false`. If you cast this float value back to `int`, you'll get an integer overflow and a negative value in return. Here's the gist:

```php
$ms = (int)($interval * 1000);
if ($ms < 0 || $interval > \PHP_INT_MAX) {
    // nope
}
return $ms;
```

Reply:
Look at this code: https://3v4l.org/WcQBR - should it output "Ok"?

Reply:
Are we talking about the boundary value? I don't really mind tbqh. We're discussing integer overflows that happen for timer intervals of round ~25 days on 32 bit platforms and millions of years on 64 bit platforms, so I don't think it's really relevant if it's +/-1 ms. What I do care about is that the outcome is consistent and reliable.

I'm not suggesting this solution is perfect, I was mostly approaching it from a different perspective because the original implementation contained an apparently arbitrary `-2` offset.

Either solution is fine for me as long as it works reliably :+1:

Reply:
I see you point of view, we could move `-2`-responsibility to user and do not bother with it in implementation - https://3v4l.org/AADvB, but we still need to do `(PHP_INT_MAX / 1000) - 2` in tests (`\React\Tests\EventLoop\AbstractLoopTest::testTimerIntervalCanBeFarInFuture`) - maybe with extensive comment about `-2`-issue.
On the other hand i'm completely fine with leaving current implementation as is, so it is up to you to choose the best for you/project/users/etc.

Reply:
The maximum working interval is `(int) (\PHP_INT_MAX / 1000) - 1`. If the value is higher, you can simply reject it before doing any conversion.

Reply:
Done.